lib/baseFactory.js

```diff
@@ -223,6 +223,48 @@ class BaseFactory {
         });
     }
 
+    /**
+     * Run raw query
+     * @method query
+     * @param  {Object}         [config]
+     * @param  {Array<Object>}  [config.queries]      Map of database type to query
+     * @param  {Object}         [config.replacements] Parameters to replace in the query
+     * @param  {Boolean}        [config.rawResponse]  Return raw response without binding to model
+     * @param  {Boolean}        [config.readOnly]     Use readOnly datastore
+     * @return {Promise}
+     */
+    query(config) {
+        const queryConfig = {
+            table: this.table,
+            queries: hoek.reach(config, 'queries'),
+            replacements: hoek.reach(config, 'replacements', { default: {} }),
+            rawResponse: hoek.reach(config, 'rawResponse', { default: false })
+        };
+
+        let datastoreForLookup = this.datastore;
+        // use read only datastore if it is configured
+        const readOnly = hoek.reach(config, 'readOnly', { default: false });
+
+        if (readOnly) {
+            datastoreForLookup = this.datastoreRO;
+        }
+
+        return datastoreForLookup.query(queryConfig).then(data => {
+            let toReturn = data;
```

Comment:
maybe a better var name than `toReturn`?

Reply:
What do you think of something like 'formattedData'?

Reply:
Good call @tkyi 
Yeah, @InderH that `formattedData` would make more sense.